File: PPavlov.Portfolio/PPavlov.Portfolio.Web.API/Models/Output/MediaOutputModel.cs
using PPavlov.Portfolio.DAL.Entities;

namespace PPavlov.Portfolio.Web.API.Models.Output
{
    public class MediaOutputModel : Media
    {
        public int Size
        {
            get
            {
                return (this.Width * this.Height) * 3;
            }
        }

        public string Src
        {
            get
            {
                return System.IO.Path.Combine(this.Url, this.Name);
            }
        }

        public static MediaOutputModel FromEntity(Media media)
        {
            return new MediaOutputModel()
            {
                Id = media.Id,
                Url = media.Url,
                Name = media.Name,
                Extension = media.Extension,
                Height = media.Height,
                Width = media.Width,
                CreatedAt = media.CreatedAt,
                UpdatedAt = media.UpdatedAt
            };
        }
    }
}

File: PPavlov.FrontEnd/src/app/areas/admin/pages/media/models/media-output-model.ts
import { LibraryOutputModel } from './library-output-model';

export interface MediaOutputModel {
    size: number;
    src: string;
    id: number;
    name: string;
    extension: string;
    url: string;
    width: number;
    height: number;
    libraryId: number;
    library?: LibraryOutputModel;
    createdAt: Date;
    updatedAt: Date;
}
